README.md

```diff
@@ -1,10 +1,70 @@
+
+By default, SwiftPM will build a debug version of the application. Note that debug versions are not suitable for running in production as they are significantly slower. To build a release version of your app, with debug symbols, run `swift build -c release -Xswiftc -g`
+
+Binary artifacts, that could be deployed, are be found under .build/x86_64-unknown-linux, or .build/x86_64-apple-macosx for macOS binaries.
+
+### Building for production
+
+- Build production code in release mode by compiling with `swift build -c release`. Running code compiled in debug mode will hurt performance signficantly. 
+
+- For best performance in Swift 5.2 or later, pass `-Xswiftc -cross-module-optimization` (this won't work in Swift versions before 5.2)
+
+- For versions prior to Swift 5.2, always build with `-Xswiftc -g` to get debugging symbols. Otherwise your stacktraces will lack most symbol names.
+
+- Integrate [`swift-backtrace`](https://github.com/swift-server/swift-backtrace) into your application to make sure backtraces are printed on crash. Backtraces do not work out-of-the-box on Linux, and this library helps to fill the gap. Eventually this will become a language feature and not require a discrete library.
+
+## Testing 
+
+Triggering the test action in Xcode or running swift test from the terminal will trigger the unit tests.
+
+SwiftPM is integrated with XCTests, Apple’s unit test framework. Test results are available in Xcode, or printed out to the terminal.
+
+Like building on Linux, testing on linux requires the use of Docker. For example:
+
+`$ docker run -v $PWD:/code -w /code swift:5.1 swift test`
+
+The above command will run the tests using the latest Swift 5.1 Docker image, utilizing bind mounts to the sources on your Mac.
+
+Swift supports architecture-specific code. By default, Foundation imports architecture-specific libraries like Darwin or Glibc. While developing on your Mac, you may end up using APIs that are not available on Linux. Since you are most likely to deploy a cloud service on Linux, it is critical to test on Linux.
+
+Another important detail about testing for Linux is the `Tests/LinuxMain.swift` file. For versions prior to Swift 5.1, This file provides SwiftPM an index of all the tests it needs to run on Linux. 
+
+- For version Swift 5.1 and later, this file is no longer required and testing with `--enable-test-discovery` seamlessly discovers the tests. This will be the default behvior in future releases of Swift.
+
+- For versions prior to Swift 5.1, it is critical to keep this file up-to-date as you add more unit tests. To regenerate this file, run `swift test --generate-linuxmain` after adding tests. It is also a good idea to include this command as part of the continuous integration setup. For version Swift 5.1 or later, this 
+
+### Testing for production
+
+- For versions Swift 5.1 and later, always test with `--enable-test-discovery` to avoid forgeting tests on Linux.
```

Comment:
I'd add

```
To remind users to add the `--enable-test-discovery` flag if they forgot, please add a file called `Tests/LinuxMain.swift` with the following content:

    fatalError("Please use `swift test --enable-test-discovery` to run the tests instead")

```

Reply:
@tomerd / @tanner0101 wdyt? I think that's a massive usability improvement if we just tell the users what they need to do there

Reply:
proof that folks are confused: https://discordapp.com/channels/431917998102675485/519613337638797315/680069804333596694

Reply:
good idea @weissi: https://github.com/swift-server/guides/pull/2/commits/243e11524238b03fa1eab98dcc6f087eca62286f

Reply:
+1